openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -1037,4 +1044,51 @@ private static SemanticModel getSemanticModel(Path clientPath) throws ProjectExc
             throw new ProjectException(e.getMessage());
         }
     }
+
+    public static String getOpenAPIType(Schema<?> schema) {
+        if (schema.getTypes() != null && !schema.getTypes().isEmpty()) {
+            for (String type : schema.getTypes()) {
+                // this returns the first non-null type in the list
+                if (!type.equals(NULL)) {
+                    return type;
+                }
+            }
+        } else if (schema.getType() != null) {
+            return schema.getType();
+        }
+        return null;
+    }
+
+    public static boolean isaArraySchema(Schema schema) {
```

Comment:
Shall we follow the following naming convention for all the newly added utils below?
```suggestion
    public static boolean isArraySchema(Schema schema) {
```

Reply:
Okay. Just thought in ER diagram IS-A relationship mindset.

Reply:
Updated